include/Support/Pipeline.h

```diff
@@ -139,6 +139,37 @@ struct DescriptorSet {
   llvm::SmallVector<Resource> Resources;
 };
 
+namespace dx {
+enum class RootParamKind {
```

Comment:
will `RootParamKind` ever grow beyond these three? The switch statements that use it today have no default cases for error handling. I'm assuming this is a non issue if you have WError on that can alter an implementer to new cases that need to be handled?

Reply:
I don't think this will grow (the enum currently covers everything D3D supports). The LLVM convention is to rely on warnings for unhandled switch cases to identify places that need updating when enums are expanded, so I think this code is correct.